homeassistant/components/climate/intesishome.py

```diff
@@ -0,0 +1,350 @@
+"""
+Support for IntesisHome Smart AC Controllers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.intesishome/
+"""
+
+import logging
+from datetime import timedelta
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (CONF_PASSWORD, CONF_USERNAME, CONF_STRUCTURE)
+from homeassistant.util import Throttle
+from homeassistant.components import persistent_notification
+from homeassistant.components.climate import (ClimateDevice,
+                                              PLATFORM_SCHEMA,
+                                              ATTR_TEMPERATURE,
+                                              ATTR_OPERATION_MODE)
+from homeassistant.const import (TEMP_CELSIUS, STATE_UNKNOWN)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'IntesisHome'
+REQUIREMENTS = ['pyintesishome==0.4']
+STATE_FAN = 'Fan'
+STATE_HEAT = 'Heat'
+STATE_COOL = 'Cool'
+STATE_DRY = 'Dry'
+STATE_AUTO = 'Auto'
+STATE_QUIET = 'Quiet'
+STATE_LOW = 'Low'
+STATE_MEDIUM = 'Medium'
+STATE_HIGH = 'High'
+STATE_OFF = 'Off'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_STRUCTURE): vol.All(cv.ensure_list, cv.string),
+})
+
+# Return cached results if last scan time was less than this value.
+# If a persistent connection is established for the controller,
+# changes to values are in realtime.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=180)
+
+try:
+    # pylint: disable=unused-import
+    from asyncio import ensure_future
+except ImportError:
+    # Python 3.4.3 and ealier has this as async
+    # pylint: disable=unused-import
+    from asyncio import async
+    ensure_future = async
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the IntesisHome interface."""
+    from pyintesishome import IntesisHome
+
+    _user = config.get(CONF_USERNAME)
+    _pass = config.get(CONF_PASSWORD)
+
+    if 'climate_intesishome' not in hass.data:
+        hass.data['climate_intesishome'] = IntesisHome(_user, _pass, hass.loop)
+
+    intesishome = hass.data['climate_intesishome']
+    intesishome.connect()
+
+    if intesishome.error_message:
+        persistent_notification.create(hass, intesishome.error_message,
+                                       "IntesisHome Error", 'intesishome')
+    add_devices([IntesisAC(deviceid, device, hass)
+                 for deviceid, device in intesishome.get_devices().items()])
+    return True
+
+
+class IntesisAC(ClimateDevice):
+    """Representation of an IntesisHome thermostat."""
+
+    def __init__(self, deviceid, device, hass):
+        """Initialize the thermostat."""
+        _LOGGER.debug('Added climate device with state: %s', repr(device))
+        self._intesishome = hass.data.get('climate_intesishome')
```

Comment:
Either pass in `intesishome` directly as an argument to the class, or move this line to the method `async_added_to_hass`. See eg:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/binary_sensor/envisalink.py#L57